octoprint_displayprogress/__init__.py

```diff
@@ -52,17 +72,40 @@ def on_print_progress(self, storage, path, progress):
 	##~~ helpers
 
 	def _send_message(self, storage, path, progress):
+		symbol = self.__class__._get_symbol(self._settings)
+
 		message = self._settings.get(["message"]).format(progress=progress,
-		                                                 storage=storage,
-		                                                 path=path,
-		                                                 bar=self.__class__._progress_bar(progress))
+														 storage=storage,
+														 path=path,
+														 bar=self.__class__._progress_bar(progress, symbol))
 		self._printer.commands("M117 {}".format(message))
 
+	## Keep a mapping of the settings name to the actual symbols
+	## TODO: Can probably make this more pythonic, but I'm having
+	## trouble with figuring out how to access settings directly
+	## when it's a class method making the call (ie: no `self`)
 	@classmethod
-	def _progress_bar(cls, progress):
-		hashes = "#" * int(round(progress / 10))
-		spaces = " " * (10 - len(hashes))
-		return "[{}{}]".format(hashes, spaces)
+	def _get_symbol(cls, settings):
```

Comment:
Why limit the selectable symbols in the first place? I'd rather turn the `barsymbol` setting into the actual symbol value here, way less mapping back and forth necessary that way :)